Python/ceval.c

```diff
@@ -408,6 +415,16 @@ Py_MakePendingCalls(void)
     if (busy)
         return 0;
     busy = 1;
+
+    /* Python signal handler doesn't really queue a callback: it only signals
+       that a signal was received, see _PyEval_SignalReceived(). */
+    UNSIGNAL_PENDING_CALLS();
+    if (PyErr_CheckSignals() < 0) {
+        busy = 0;
```

Comment:
I suggest to move this code at the end of the function, in a new "error:" label and use goto. IMHO it's more readable like that (set busy to 1 at the function entry and set it to 0 at exit) (and it would avoid code duplication).